subsys/bluetooth/host/audio/mpl_internal.h

```diff
@@ -84,6 +84,12 @@ struct mpl_mediaplayer_t {
 #endif /* CONFIG_BT_OTS || CONFIG_BT_OTC */
 	uint8_t              content_ctrl_id;
 	struct media_proxy_pl_calls calls;
+
+	bool                        next_track_set; /* If next track explicitly set */
```

Comment:
I'm wondering if we should append a "pending" to this field to make it more clear that is a pending action, e.g. `next_track_set_pending`.

Reply:
Not sure if that is clearer.  When the next track has been set, it has been set, and is no longer pending.
There is always (with a few exceptions) a next track, and this one is no more pending than the one that would have been used if it was not set explicitly.

Reply:
I am not sure I follow, or perhaps even understand this field. 
This is some boolean that is set by a client (or the server itself perhaps as well?) to change to the next track. At some point we check this boolean, and if it is set, we change the track and clear it. 

To me that sounds like a short term state variable for a pending action that is cleared once the action has been executed, but perhaps I am misunderstanding something?

Reply:
It is not a boolean signaling to change to the next track.  It is a boolean to signal that the next track has been set to a different track than the one that would be chosen from the normal playing order.  And the struct says which track this is.
So, at the next track change, this (given) track shall become the current track, instead of the current track's "normal" next track.
There is always a next track, but in this case which track is the next track is an exception from the normal.

(This could be solved in other ways as well, but the ones I am thinking of involves a more elaborate group/track pointers, and I do not feel that is worth it now.)